xds/type/matcher/v3/range.proto

```diff
@@ -0,0 +1,58 @@
+syntax = "proto3";
+
+package xds.type.matcher.v3;
+
+import "validate/validate.proto";
+
+option java_package = "com.github.xds.type.matcher.v3";
+option java_outer_classname = "RangeProto";
+option java_multiple_files = true;
+option go_package = "github.com/cncf/xds/go/xds/type/matcher/v3";
+
+// [#protodoc-title: Range matcher]
+
+// Specifies the int64 start and end of the range using half-open interval
+// semantics [start, end).
+message Int64Range {
+  // start of the range (inclusive)
+  int64 start = 1;
+
+  // end of the range (exclusive)
+  int64 end = 2;
+}
+
+// Specifies a set of ranges for matching an int64 number.
```

Comment:
Are these sublinear matchers or linear matchers?  If sublinear, what are the precedence rules when there are multiple matching ranges?  Please document.

Reply:
First one wins. Thanks. Yes, this is intended to be implemented with a segment tree or the likes.